### DISCUSSION TITLE
변수에 값이 할당 될 때 메모리 공간 측면에서 값이 어떻게 저장되는 지 설명하세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#21] [변수]
- 변수에 값이 (재)할당 될 때 메모리 공간 측면에서 값이 어떻게 저장되는 지 설명하세요.

**Comment by pipisebastian** [comment]
>변수에 값이 할당 될 때 메모리 공간 측면에서 값이 어떻게 저장되는 지 설명하세요.

선언과 달리, 할당은 런타임 시점에서 실행됩니다. 새로운 메모리 공간을 확보하고, 값을 저장합니다. 원래 값이 저장되어있던 메모리 공간은 필요없게 되며, 이는 가비지 콜렉터에 의해 자동으로 해제됩니다.

**Comment by publdaze** [comment]
> - 변수에 값이 할당 될 때 메모리 공간 측면에서 값이 어떻게 저장되는 지 설명하세요.
- 기존에 있던 메모리 주소의 값을 변경하는 게 아닌 새로운 메모리 공간을 할당 받아 저장 후 변수의 식별자와 매핑함
- 기존의 메모리 주소 값은 참조하고 있는 변수가 없다면 가비지 콜렉터에 의해 메모리가 해제 됨
